PHP-1

avalanche123 opened the pull request:
Windows build support

Commit:
move stubs to doc

Commit:
move Cluster related classes into C API

introduce persistent clusters support
move Cluster\Builder, Cluster, DefaultCluster and SSLOptions into extension
introduce source tree for C classes

Commit:
rename number parsing functions

Commit:
fix header paths

Commit:
don't redefine classes

Commit:
Merge branch 'master' into source-reorg

Commit:
start adding support for persistent session objects

Commit:
implement ssl options builder

Commit:
fix error messages

Commit:
fix future error formatting

Commit:
set the actual log level

Commit:
close stream when done

Commit:
set client cert instead of add trusted cert

Commit:
remove deprecated functions

Commit:
move exceptions to the class hierarchy

Commit:
fix method name

Commit:
move Cassandra class into extension

Commit:
update external classes

Commit:
started simple, prepared, and batch statements

Commit:
fixed missing files from config.m4

Commit:
add missing class definition

Commit:
simple and prepared statements mostly working

Commit:
fix variable definition on php 5.3

Commit:
unquote expected message

Commit:
make cql a required param

Commit:
fix exception class entries

Commit:
fix cs and destructor

Commit:
fixes for execution options

Commit:
win: Adding Windows build configuration and script

This commit starts the Windows build process for the php-driver by
ensuring dependencies are met for building and gathers required
source for building the driver.

Currently the driver does not link properly and contains undefined
references for SPL dependency.

Commit:
added methods to batch type

Commit:
fixes for cluster connect/connectAsync

Commit:
fix globals declaration

Commit:
fix leaks when initializing session

Commit:
fix thread-safety

Commit:
add initial implementation of Cassandra\Rows

Commit:
remove unused function

Commit:
redefine assert macros

Commit:
fix macro name

Commit:
fix formatting

Commit:
update execution options usage

Commit:
added batching

Commit:
fix results pagination

Commit:
don't use persistent sessions in ccm harness

Commit:
fix tsrm

Commit:
initialize zval

Commit:
api: Upgrading to C/C++ driver v2.0.0 while fixing Windows build

This commit upgrade the submodule and code to work with the C/C++
driver v2.0.0 while addressing cross platform build issues on the
Windows platform. Cleanup was also done to remove parent relative
includes for php_cassandra.h.

Commit:
implement Cassandra\Session::close() and closeAsync()

Commit:
allow optional timeout argument to Cassandra\FutureValue::get()

Commit:
add missing FutureClose.c

Commit:
build: Fixing Windows link issues and adding details to DLL

This commit addresses the previous link issues with the Windows DLL.
The build script was also update to work from the cpp-driver submodule
and the version information was added to the details of the DLL during
compilation; version.h was created to split from main extension header.

Commit:
build: Fixing include and update Vagrant file for linux build

Added libpcre3-dev for SPL dependency

Commit:
added logging to file/syslog

Commit:
build: Minimal PHP for building and testing the driver

Enabled extensions for installing composer and required dependencies
for the driver; composer.json was updated to download PHPUnit
required dependencies. The ability to disable thread safety, enable
zlib and build in debug was added also (cpp-driver submodule was
updated).

NOTE: Using compiled 32-bit PHP results in errors during install of
required dependencies with composer (might be due to zip extension
not being enabled); the driver still builds correctly.

Commit:
removed TSRM-aware functions from the logging callback

Commit:
fixed implicit decl warnings

avalanche123 committed:
Merge pull request #22 from datastax/PHP-11

added logging to file/syslog

Commit:
build: Add PHP v5.3 and patch v5.4, remove Windows SDK, add test config

Windows SDK support has been removed due to the nature of the PHP
build process; Windows SDK is not possible. In order to create a
simple configuration process for building the driver and creating a
process to run tests in Windows, zlib is being enabled by default for
the cpp-driver and in turn the php-driver (and OpenSSL). This allows
for enabled the zip extension during the PHP build process to work
better with composer for installing dependencies.

Driver builds and additional testing configuration builds have been
split up to allow for quicker builds when just the driver is wanted.
Test configuration builds are disabled by default but can be enabled
with --ENABLE-TEST-CONFIGURATION.

Commit:
build: Added PHP v5.3 to help message

Commit:
build: Adding test configuration argument to help message

avalanche123 commented on .gitignore:
```diff
@@ -5,3 +5,4 @@
 /.vagrant/
 /venv/
 /tmp/*
+/*.swp
```
is this a vim swap file being ignored, if so, it should be included in the global gitignore on your system - https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

Commit:
build: Adding dependency to build instructions and fixing Vagrant build

Commit:
force remove shared lib during install

Commit:
fix const variables

Commit:
fix include path

Commit:
fix clang warnings

Commit:
fix bind function

Commit:
add -fPIC compiler flags

Commit:
fix env var

Commit:
Merge branch 'master' into PHP-1

Commit:
Merge remote-tracking branch 'origin/PHP-12' into PHP-1-merge-PHP-12

Commit:
upgrade to new cpp driver api

avalanche123 committed:
Merge pull request #24 from datastax/PHP-1-merge-PHP-12

Merge PHP-12 into PHP-1

Commit:
fix includes

Commit:
make default paging size, consistency level and timeout work

Commit:
win: Fix build, add key generate script and fix CCM bridge

Commit:
build: Force ANSI C90 and Fix up Linux and Windows Build

This commit forces ANSI C90 for GCC and Clang compilers by issuing
the -pedantic-errors when compiling; ease cross platform compilations.
CFLAGS and LIBS have been moved from the install and Vagrant scripts
to the config,m4 for *nix builds (some clean up is still necessary).

NOTE: The Windows build does not properly link currently.

avalanche123 commented on ext/config.m4:
```diff
@@ -7,28 +7,92 @@ if test -z "$PHP_GMP"; then
+      CASSANDRA_CFLAGS="-Wall -pedantic -Wextra -Wno-long-long -Wno-deprecated-declarations -Wno-unused-parameter -Wno-variadic-macros"
+      ;;
+  esac
+  CASSANDRA_LIBS="-lssl -lz -luv -lstdc++"
```
why does the driver depend on stdc++ ?

mikefero replied:
This was taken from the install.sh and Vagrantfile scripts where the LIBS were supplied, but it should be for the cpp-driver.

mikefero replied:
This is where some cleanup should be done and some additional library checks as well since we are relying on things for the php-driver.

avalanche123 replied:
Right, I agree that we need to add checks for libuv since it is used in the driver, but we don't use any of the stdc++ things directly in the driver, that's why those flags were supplied from the outside for static linking the cpp-driver only

mikefero replied:
Reverted the changes back to the way they were. I still feel that all these build settings should be in the config.m4 and not be required outside the build like it is currently.

Commit:
build: Fixing Travis Build

Commit:
win: Remove strndup, disable syslog and add build instructions

mikefero commented:
strndup is still used in php_cassandra.c (https://github.com/datastax/php-driver/blob/PHP-1/ext/php_cassandra.c#L656) which is why I added it back in ... will update to use estrndup.

avalanche123 commented:
Ah I thought I've removed it already, definitely feel free to change that

Commit:
fix: Debug on Windows, Vagrant, and remove pedantic-errors

pedantic-errors flag was removed from *nix compilers and replaced
with -pedantic due to issues with enumarations in PHP headers not
conforming to the standard properly.

Commit:
fix: Windows thread hang on PHP shutdown

NOTE: This may not work properly with ZTS since multiple threads
may be operating on the logger while another is shutting it down
inside the module shutdown.

Commit:
Cleaning up the Windows SSL key generate script

Commit:
win: Correct build issues with VS2010 for PHP v5.3

This commit allows for the dirver to build against PHP v5.3. A small
fix in the build script was also made to ensure PHP dependencies were
copied correctly and that cpp-driver options were targeted as they
should be.

The cpp-driver submodule was also updated to fix an issue when
building with Boost atomic (PHP driver build script forces this
setting for VS2010 builds).
